src/main/java/com/example/clone/service/UserInfoService.java

```diff
@@ -0,0 +1,50 @@
+package com.example.clone.service;
+
+import com.example.clone.model.UserInfo;
+import com.example.clone.repository.UserInfoRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class UserInfoService implements UserDetailsService {
+
+    @Autowired
+    private UserInfoRepository repository;
+
+    @Autowired
+    private PasswordEncoder encoder;
```

Comment:
Which implementation of PasswordEncoder are you using here?

Reply:
Im using bcrypt.